src/pages/home/report/ReportActionCompose.js

```diff
@@ -204,7 +224,55 @@ class ReportActionCompose extends React.Component {
     }
 
     onSelectionChange(e) {
-        this.setState({selection: e.nativeEvent.selection});
+        const nextChar = this.state.value.charAt(e.nativeEvent.selection.end);
+
+        const leftString = this.state.value.substring(
+            0,
+            e.nativeEvent.selection.end,
+        );
+        const colonIndex = leftString.lastIndexOf(':');
+
+        const numberOfHighlightedLines = this.state.value
+            .substr(0, e.nativeEvent.selection.end)
+            .split('\n').length;
+
+        const isCurrentlyShowingEmojiSuggestion = this.isLikeEmojiCode(
+            this.state.value,
+            e.nativeEvent.selection.end,
+        );
+
+        // If the window width narrow, probably a mobile device
+        const isMobileDeviceScreen = !(this.props.isMediumScreenWidth || this.props.isLargeScreenWidth);
```

Comment:
This seems simpler? But also, I think it means we can get rid of this variable altogether and just use `this.props.isSmallScreenWidth` instead.

```suggestion
        const isMobileDeviceScreen = this.props.isSmallScreenWidth;
```